File: Interop.cs
using System.Runtime.InteropServices.JavaScript;
using System.Runtime.Versioning;

namespace BlazorWASM70UnmarshalledJS.JavaScriptInterop
{
    [SupportedOSPlatform("browser")]
    public partial class Interop
    {
        [JSImport("getMessage", "Interop")]
        internal static partial string GetWelcomeMessage();

        [JSImport("setMessage", "Interop")]
        internal static partial void SetWelcomeMessage();

        [JSExport]
        internal static string GetMessageFromDotnet()
        {
            return "Olá do Blazor!";
        }
    }
}

File: wwwroot/js/interop.js
export function getMessage() {
    return 'Olá do Blazor!';
}

export async function setMessage() {
    const { getAssemblyExports } = await globalThis.getDotnetRuntime(0);
    var exports = await getAssemblyExports("BlazorWASM70UnmarshalledJS.dll");

    document.getElementById("result").innerText =
        exports.BlazorWASM70UnmarshalledJS.JavaScriptInterop.Interop.GetMessageFromDotnet();
}
